### PR TITLE
add runner3680

### PR DESCRIPTION
**Pull request by omkbd**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* add runner3680 keyboard

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/runner3680/3x6/3x6.c [hunk 1]
```diff
@@ -0,0 +1 @@
+#include "runner3680.h"
```
Comment (drashna) [inline keyboards/runner3680/3x6/3x6.c]: This should match the folder it's in. Eg: 

```suggestion
#include "3x6.h"
```

File: keyboards/runner3680/4x6/config.h [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+// #define MATRIX_COL_PINS { B6, B2, B3, B1, F7, F6} //uncomment this line and comment line above if you need to reverse left-to-right key order
+
+/* define tapping term */
+#define TAPPING_TERM 120
+
+/* define if matrix has ghost */
+//#define MATRIX_HAS_GHOST
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (drashna) [inline keyboards/runner3680/4x6/config.h]: `DEBOUNCING_DELAY` is deprecated and should be changed

```suggestion
#define DEBOUNCE 5
```

File: keyboards/runner3680/4x7/config.h [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+// #define MATRIX_COL_PINS { B6, B2, B3, B1, F7, F6, F5 } //uncomment this line and comment line above if you need to reverse left-to-right key order
+
+/* define tapping term */
+#define TAPPING_TERM 120
+
+/* define if matrix has ghost */
+//#define MATRIX_HAS_GHOST
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (drashna) [inline keyboards/runner3680/4x7/config.h]: `DEBOUNCING_DELAY` is deprecated and should be changed

```suggestion
#define DEBOUNCE 5
```

File: keyboards/runner3680/4x8/config.h [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+// #define MATRIX_COL_PINS { B6, B2, B3, B1, F7, F6, F5, F4 } //uncomment this line and comment line above if you need to reverse left-to-right key order
+
+/* define tapping term */
+#define TAPPING_TERM 120
+
+/* define if matrix has ghost */
+//#define MATRIX_HAS_GHOST
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (drashna) [inline keyboards/runner3680/4x8/config.h]: `DEBOUNCING_DELAY` is deprecated and should be changed

```suggestion
#define DEBOUNCE 5
```

File: keyboards/runner3680/5x7/config.h [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+// #define MATRIX_COL_PINS { B6, B2, B3, B1, F7, F6, F5 } //uncomment this line and comment line above if you need to reverse left-to-right key order
+
+/* define tapping term */
+#define TAPPING_TERM 120
+
+/* define if matrix has ghost */
+//#define MATRIX_HAS_GHOST
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (drashna) [inline keyboards/runner3680/5x7/config.h]: `DEBOUNCING_DELAY` is deprecated and should be changed

```suggestion
#define DEBOUNCE 5
```

File: keyboards/runner3680/5x8/config.h [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+// #define MATRIX_COL_PINS { B6, B2, B3, B1, F7, F6, F5, F4 } //uncomment this line and comment line above if you need to reverse left-to-right key order
+
+/* define tapping term */
+#define TAPPING_TERM 120
+
+/* define if matrix has ghost */
+//#define MATRIX_HAS_GHOST
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (drashna) [inline keyboards/runner3680/5x8/config.h]: `DEBOUNCING_DELAY` is deprecated and should be changed

```suggestion
#define DEBOUNCE 5
```

File: keyboards/runner3680/runner3680.h [hunk 7]
```diff
@@ -0,0 +1,42 @@
+#ifndef RUNNER3680_H
```
Comment (drashna) [inline keyboards/runner3680/runner3680.h]:
```suggestion
#pragma once 
```

File: keyboards/runner3680/runner3680.h [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+#define RUNNER3680_H
```
Comment (drashna) [inline keyboards/runner3680/runner3680.h]:
```suggestion
```

File: keyboards/runner3680/runner3680.h [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+#ifdef KEYBOARD_runner3680_3x7
+  #include "3x7.h"
+#endif
+
+#ifdef KEYBOARD_runner3680_3x6
+  #include "3x6.h"
+#endif
+
+#endif
```
Comment (drashna) [inline keyboards/runner3680/runner3680.h]:
```suggestion
```

File: keyboards/runner3680/4x6/4x6.h [hunk 10]
```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "../runner3680.h"
+
+#include "quantum.h"
+
+#ifdef USE_I2C
```
Comment (drashna) [inline keyboards/runner3680/4x6/4x6.h]: I don't think that this block is needed, and can be removed.

File: keyboards/runner3680/4x6/4x6.h [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+#include <stddef.h>
+#ifdef __AVR__
+	#include <avr/io.h>
+	#include <avr/interrupt.h>
+#endif
+#endif
+
+#define LAYOUT( \
```
Comment (drashna) [inline keyboards/runner3680/4x6/4x6.h]:
```suggestion
#define LAYOUT_ortho_4x12( \
```

File: keyboards/runner3680/4x6/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+     * | Tab  |   Q  |   W  |   E  |   R  |   T  |      |   Y  |   U  |   I  |   O  |   P  |   [  |
+     * |------+------+------+------+------+------|      |------+------+------+------+------+------|
+     * | Esc  |   A  |   S  |   D  |   F  |   G  |      |   H  |   J  |   K  |   L  |   ;  |   "  |
+     * |------+------+------+------+------+------|      |------+------+------+------+------+------|
+     * | Shift|   Z  |   X  |   C  |   V  |   B  |      |   N  |   M  |   ,  |   .  |   /  |   \  |
+     * |------+------+------+------+------+------|      |------+------+------+------+------+------|
+     * | Ctrl | GUI  | Alt  | EISU |  Del | Space|      | Enter| Bksp | EISU |  ESC |Adjust| Left |
+     * `-----------------------------------------'      `-----------------------------------------'
+     */
+  [_QWERTY] = LAYOUT( \
```
Comment (drashna) [inline keyboards/runner3680/4x6/keymaps/default/keymap.c]:
```suggestion
  [_QWERTY] = LAYOUT_ortho_4x12( \
```

File: keyboards/runner3680/4x6/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+     * |      |RGBRST| RESET|      |      |      |      |      |      |      |      |      |      |
+     * |------+------+------+------+------+------|      |------+------+------+------+------+------|
+     * |      |  TOG |  HUI |  SAI |  VAI |      |      |      |      |      |      |      |      |
+     * |------+------+------+------+------+------|      |------+------+------+------+------+------|
+     * |      |  MOD |  HUD |  SAD |  VAD |      |      |      |      |      |      |      |      |
+     * |------+------+------+------+------+------|      |------+------+------+------+------+------|
+     * |      |      |      |      |      |      |      |      |      |      |      |      |      |
+     * `-----------------------------------------'      `-----------------------------------------'
+     */
+  [_ADJUST] =  LAYOUT( \
```
Comment (drashna) [inline keyboards/runner3680/4x6/keymaps/default/keymap.c]:
```suggestion
  [_ADJUST] =  LAYOUT_ortho_4x12( \
```

File: keyboards/runner3680/4x6/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,105 @@
+#include QMK_KEYBOARD_H
+#include "action_layer.h"
```
Comment (drashna) [inline keyboards/runner3680/4x6/keymaps/default/keymap.c]: Could you remove all of these additional includes?  

```suggestion
```

They get included automatically, at various points in the built process, so they are redundant and not necessary here.

File: keyboards/runner3680/4x6/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+#include "eeconfig.h"
```
Comment (drashna) [inline keyboards/runner3680/4x6/keymaps/default/keymap.c]: Could you remove all of these additional includes?  

```suggestion
```

They get included automatically, at various points in the built process, so they are redundant and not necessary here.

File: keyboards/runner3680/4x6/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+};
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+  ADJUST,
+  RGBRST
+};
+
+// Fillers to make layering more clear
+#define _______ KC_TRNS
```
Comment (drashna) [inline keyboards/runner3680/4x6/keymaps/default/keymap.c]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/runner3680/4x6/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+#define XXXXXXX KC_NO
```
Comment (drashna) [inline keyboards/runner3680/4x6/keymaps/default/keymap.c]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/runner3680/4x6/keymaps/default/keymap.c [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+     */
+  [_ADJUST] =  LAYOUT( \
+      _______, RGBRST,  RESET,   _______, _______, _______,    _______, _______, _______, _______, _______, _______, \
+      _______, RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI, _______,    _______, _______, _______, _______, _______, _______, \
+      _______, RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD, _______,    _______, _______, _______, _______, _______, _______, \
+      _______, _______, _______, _______, _______, _______,    _______, _______, _______, _______, _______, _______  \
+      )
+};
+
+void persistent_default_layer_set(uint16_t default_layer) {
```
Comment (drashna) [inline keyboards/runner3680/4x6/keymaps/default/keymap.c]: This block isn't used and should be removed.

File: keyboards/runner3680/4x7/4x7.c [hunk 19]
```diff
@@ -0,0 +1 @@
+#include "runner3680.h"
```
Comment (drashna) [inline keyboards/runner3680/4x7/4x7.c]:
```suggestion
#include "4x7.h"
```

File: keyboards/runner3680/4x7/4x7.h [hunk 20]
```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "../runner3680.h"
```
Comment (drashna) [inline keyboards/runner3680/4x7/4x7.h]:
```suggestion
#include "runner3680.h"
```

File: keyboards/runner3680/4x7/keymaps/default/keymap.c [hunk 21]
```diff
@@ -0,0 +1,105 @@
+#include QMK_KEYBOARD_H
+#include "action_layer.h"
```
Comment (drashna) [inline keyboards/runner3680/4x7/keymaps/default/keymap.c]: Could you remove all of these additional includes?  

```suggestion
```

They get included automatically, at various points in the built process, so they are redundant and not necessary here.

File: keyboards/runner3680/4x7/keymaps/default/keymap.c [hunk 22]
```diff
@@ -0,0 +1,105 @@
...
+#include "eeconfig.h"
```
Comment (drashna) [inline keyboards/runner3680/4x7/keymaps/default/keymap.c]: Could you remove all of these additional includes?  

```suggestion
```

They get included automatically, at various points in the built process, so they are redundant and not necessary here.

File: keyboards/runner3680/4x7/keymaps/default/keymap.c [hunk 23]
```diff
@@ -0,0 +1,105 @@
...
+};
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+  ADJUST,
+  RGBRST
+};
+
+// Fillers to make layering more clear
+#define _______ KC_TRNS
```
Comment (drashna) [inline keyboards/runner3680/4x7/keymaps/default/keymap.c]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/runner3680/4x7/keymaps/default/keymap.c [hunk 24]
```diff
@@ -0,0 +1,105 @@
...
+#define XXXXXXX KC_NO
```
Comment (drashna) [inline keyboards/runner3680/4x7/keymaps/default/keymap.c]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/runner3680/4x7/keymaps/default/keymap.c [hunk 25]
```diff
@@ -0,0 +1,105 @@
...
+     */
+  [_ADJUST] =  LAYOUT( \
+      _______, _______, RGBRST,  RESET,   _______, _______, _______,    _______, _______, _______, _______, _______, _______, _______, \
+      _______, _______, RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI, _______,    _______, _______, _______, _______, _______, _______, _______, \
+      _______, _______, RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD, _______,    _______, _______, _______, _______, _______, _______, _______, \
+      _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______, _______, _______, _______, _______  \
+      )
+};
+
+void persistent_default_layer_set(uint16_t default_layer) {
```
Comment (drashna) [inline keyboards/runner3680/4x7/keymaps/default/keymap.c]: This block isn't used, and should be removed,

File: keyboards/runner3680/4x8/4x8.c [hunk 26]
```diff
@@ -0,0 +1 @@
+#include "runner3680.h"
```
Comment (drashna) [inline keyboards/runner3680/4x8/4x8.c]:
```suggestion
#include "4x8.h"
```

File: keyboards/runner3680/4x8/4x8.h [hunk 27]
```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "../runner3680.h"
```
Comment (drashna) [inline keyboards/runner3680/4x8/4x8.h]:
```suggestion
#include "runner3680.h"
```

File: keyboards/runner3680/4x8/keymaps/default/keymap.c [hunk 28]
```diff
@@ -0,0 +1,105 @@
...
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+  ADJUST,
+  RGBRST
+};
+
+// Fillers to make layering more clear
+#define _______ KC_TRNS
+#define XXXXXXX KC_NO
```
Comment (drashna) [inline keyboards/runner3680/4x8/keymaps/default/keymap.c]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/runner3680/4x8/keymaps/default/keymap.c [hunk 29]
```diff
@@ -0,0 +1,105 @@
...
+};
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+  ADJUST,
+  RGBRST
+};
+
+// Fillers to make layering more clear
+#define _______ KC_TRNS
```
Comment (drashna) [inline keyboards/runner3680/4x8/keymaps/default/keymap.c]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/runner3680/4x8/keymaps/default/keymap.c [hunk 30]
```diff
@@ -0,0 +1,105 @@
+#include QMK_KEYBOARD_H
+#include "action_layer.h"
+#include "eeconfig.h"
```
Comment (drashna) [inline keyboards/runner3680/4x8/keymaps/default/keymap.c]: Could you remove all of these additional includes?  

```suggestion
```

They get included automatically, at various points in the built process, so they are redundant and not necessary here.

File: keyboards/runner3680/4x8/keymaps/default/keymap.c [hunk 31]
```diff
@@ -0,0 +1,105 @@
+#include QMK_KEYBOARD_H
+#include "action_layer.h"
```
Comment (drashna) [inline keyboards/runner3680/4x8/keymaps/default/keymap.c]: Could you remove all of these additional includes?  

```suggestion
```

They get included automatically, at various points in the built process, so they are redundant and not necessary here.

File: keyboards/runner3680/4x8/keymaps/default/keymap.c [hunk 32]
```diff
@@ -0,0 +1,105 @@
...
+     */
+  [_ADJUST] =  LAYOUT( \
+      _______, _______, _______, RGBRST,  RESET,   _______, _______, _______,    _______, _______, _______, _______, _______, _______, _______, _______,\
+      _______, _______, _______, RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI, _______,    _______, _______, _______, _______, _______, _______, _______, _______,\
+      _______, _______, _______, RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD, _______,    _______, _______, _______, _______, _______, _______, _______, _______,\
+      _______, _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______, _______, _______, _______, _______, _______ \
+      )
+};
+
+void persistent_default_layer_set(uint16_t default_layer) {
```
Comment (drashna) [inline keyboards/runner3680/4x8/keymaps/default/keymap.c]: This block is unused, and should be removed.

File: keyboards/runner3680/5x6/5x6.c [hunk 33]
```diff
@@ -0,0 +1 @@
+#include "runner3680.h"
```
Comment (drashna) [inline keyboards/runner3680/5x6/5x6.c]:
```suggestion
#include "5x6.h"
```

File: keyboards/runner3680/5x6/5x6.h [hunk 34]
```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "../runner3680.h"
```
Comment (drashna) [inline keyboards/runner3680/5x6/5x6.h]:
```suggestion
#include "runner3680.h"
```

File: keyboards/runner3680/5x6/config.h [hunk 35]
```diff
@@ -0,0 +1,67 @@
...
+// #define MATRIX_COL_PINS { B6, B2, B3, B1, F7, F6} //uncomment this line and comment line above if you need to reverse left-to-right key order
+
+/* define tapping term */
+#define TAPPING_TERM 120
+
+/* define if matrix has ghost */
+//#define MATRIX_HAS_GHOST
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (drashna) [inline keyboards/runner3680/5x6/config.h]: `DEBOUNCING_DELAY` is deprecated and should be changed

```suggestion
#define DEBOUNCE 5
```

File: keyboards/runner3680/5x6/keymaps/default/keymap.c [hunk 36]
```diff
@@ -0,0 +1,111 @@
+#include QMK_KEYBOARD_H
+#include "action_layer.h"
```
Comment (drashna) [inline keyboards/runner3680/5x6/keymaps/default/keymap.c]: Could you remove all of these additional includes?  

```suggestion
```

They get included automatically, at various points in the built process, so they are redundant and not necessary here.

File: keyboards/runner3680/5x6/keymaps/default/keymap.c [hunk 37]
```diff
@@ -0,0 +1,111 @@
...
+#include "eeconfig.h"
```
Comment (drashna) [inline keyboards/runner3680/5x6/keymaps/default/keymap.c]: Could you remove all of these additional includes?  

```suggestion
```

They get included automatically, at various points in the built process, so they are redundant and not necessary here.

File: keyboards/runner3680/3x6/keymaps/default/keymap.c [hunk 38]
```diff
@@ -0,0 +1,99 @@
...
+    /* QWERTY
+     * ,-----------------------------------------.      ,-----------------------------------------.
+     * | Tab  |   Q  |   W  |   E  |   R  |   T  |      |   Y  |   U  |   I  |   O  |   P  |   [  |
+     * |------+------+------+------+------+------|      |------+------+------+------+------+------|
+     * | Esc  |   A  |   S  |   D  |   F  |   G  |      |   H  |   J  |   K  |   L  |   ;  |   "  |
+     * |------+------+------+------+------+------|      |------+------+------+------+------+------|
+     * | Shift|   Z  |   X  |   C  |   V  |   B  |      |   N  |   M  |   ,  |   .  |   /  |   \  |
+     * `-----------------------------------------'      `-----------------------------------------'
+     */
+  [_QWERTY] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/runner3680/3x6/keymaps/default/keymap.c]: The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater and less "hack-y".

File: keyboards/runner3680/3x6/keymaps/default/keymap.c [hunk 39]
```diff
@@ -0,0 +1,99 @@
...
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+      case QWERTY:
+        if (record->event.pressed) {
+           print("mode just switched to qwerty and this is a huge string\n");
+          set_single_persistent_default_layer(_QWERTY);
+        }
+        return false;
+        break;
```
Comment (fauxpark) [inline keyboards/runner3680/3x6/keymaps/default/keymap.c]: `break` after `return` is dead code.

Comment (drashna) [reply keyboards/runner3680/3x6/keymaps/default/keymap.c]: true, best to get rid of the return.

**Pull request merged**